### PR TITLE
[minor] redo docs, remove outdated parts and add links

### PR DESCRIPTION
**Pull request by Swaagie**
## Summary

- Update minor/patch version of `tar-fs`  other security warnings are related to `cdnup` using an older `pkgcloud` and usage of `babel@5` to do ES6 compiling.
- Redo README.md
- Add badges
- Remove API `cancel` route, it does not exist anymore.
- Use generic collected docs
- Add links to NSQ, carpenterd-worker, babel, webpack, browserify, etc.

## Changelog

Updated

## Test Plan

NA

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,103 +1,40 @@
...
+[![License](https://img.shields.io/npm/l/carpenterd.svg?style=flat-square)](https://github.com/godaddy/carpenterd/blob/master/LICENSE)
+[![npm Downloads](https://img.shields.io/npm/dm/carpenterd.svg?style=flat-square)](https://npmcharts.com/compare/carpenterd?minimal=true)
+[![Build Status](https://travis-ci.org/godaddy/carpenterd.svg?branch=master)](https://travis-ci.org/godaddy/carpenterd)
+[![Dependencies](https://img.shields.io/david/godaddy/carpenterd.svg?style=flat-square)](https://github.com/godaddy/carpenterd/blob/master/package.json)
 
-##### Build systems
+Build and compile npm packages to run in the browser. This API is capable of
+building modules through different build systems. The aim is to have full
+cross-build-system API that serves a single file to be used in the browser.
+Note that this API should only be hit from the [`warehouse.ai`][warehouse.ai].
```
Comment (msluther) [inline README.md:12]: nit: remove the `the`

> Note that this API should only be hit from [`warehouse.ai`][warehouse.ai].

File: README.md [hunk 2]
```diff
@@ -154,33 +91,107 @@ Host: localhost:6064
...
+defaults from Carpenters configuration.
+
+**type:** can be supplied as `build` property on the package.json or is
+extracted from the keywords. Defaults to ES6.
+
+**target:** writes the package and its dependencies to a temporary folder
+named after `build.id` a unique v4 `id`. After building this folder is
+removed from the file system to save disk space.
+
+**env:** Retrieved from the package.json `env` property.
```
Comment (msluther) [inline README.md:180]: that's not true is it? this is based on the dist-tag / what environment it's built for, right?

File: README.md [hunk 3]
```diff
@@ -154,33 +91,107 @@ Host: localhost:6064
...
+
+**version:** read from the package.json `dist-tags.latest`. Has no default.
+
+**name:** Defaults to the package.json `name` property, e.g. the modules name.
+
+**locale:** Uses the locales specified on the `package.json`. Each unique
```
Comment (msluther) [inline README.md:186]: > specified on the `package.json`.

There's going to be a lot to come back through here to clean up once the [configuration changes](https://github.com/warehouseai/extract-config/) are done. cc: @SivanMehta

**Review by msluther — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (jcrugzz) [reply README.md:180]: yea its not set on package.json, its just something we kind of add to the data structure of the publish payload or the tarball that we fetch and base64 into the same data structure.

Comment (Swaagie) [reply README.md:180]: Nice to see how we catch discrepancies like this. When I first read it I was also like, really?!? 👍 😄

**Pull request merged**